src/app/downloads/downloads.component.ts

```diff
@@ -2,26 +2,26 @@ import { Component, OnInit, OnDestroy } from '@angular/core';
 import { Router } from '@angular/router';
 import { Store } from '@ngrx/store';
 import { Subscription } from 'rxjs/Subscription';
+import { Observable } from 'rxjs/Observable';
 import { CastleService } from '../core';
-import { CastlePodcastMetricsAction, CastleEpisodeMetricsAction,
-  GoogleAnalyticsEventAction, CastleEpisodeChartToggleAction } from '../ngrx/actions';
+import * as ACTIONS from '../ngrx/actions';
 import { RouterModel, EpisodeModel, PodcastModel, ChartType, MetricsType,
   CHARTTYPE_PODCAST, INTERVAL_DAILY, EPISODE_PAGE_SIZE, METRICSTYPE_DOWNLOADS, getMetricsProperty } from '../ngrx';
-import { selectRouter, selectEpisodes, selectPodcasts } from '../ngrx/reducers';
+import { selectRouter, selectEpisodes, selectPodcasts, selectLoading, selectLoaded, selectErrors } from '../ngrx/reducers';
 import { filterPodcastEpisodePage } from '../shared/util/metrics.util';
 import * as dateUtil from '../shared/util/date';
 import { isPodcastChanged, isBeginDateChanged, isEndDateChanged, isIntervalChanged } from '../shared/util/filter.util';
 
 @Component({
   template: `
-    <prx-spinner *ngIf="isPodcastLoading || isEpisodeLoading" overlay="true"
+    <prx-spinner *ngIf="loading$ | async" overlay="true"
```

Comment:
I don't have any hard metrics on this at all, but seems like this was also a good move for improving browser performance. My fan isn't so loud anymore when the page is loading. There was a lot of data being filtered and worked through and page draws and charts generated every time a new piece of data came in.